MainActivity.java

```diff
@@ -69,6 +69,35 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
         });
     }
 
+   /* 
+    The function everytime the activity is started
+   */
+
+    @Override
+    protected void onStart()
+    {
+        super.onStart();
+        if (!IsEmailVerified()) {
+            showEmailVerificationPopUp = new Intent(Mainpage.this, emailverificationpopup.class);
+            startActivity(showEmailVerificationPopUp);
+        }
+
+    }
+	
+    /*
+    The function check if the user wither newly registered or existing user has verified email or not and then send the intent accordingly
+    */	
+    private boolean IsEmailVerified() {
+
+        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
+        if (user != null) {
+            return user.isEmailVerified();   // check if user is verified or not
+        }
+        else
+            return false;
+    }
+

```

Comment:
These are newly added lines into the code , review it and if there are no issues kindly merge and commit